packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -1086,13 +1086,18 @@ export const ComboBoxMixin = (subclass) =>
 
     /** @private */
     _detectAndDispatchChange() {
+      const isValueChanged = this.value !== this._lastCommittedValue;
+      if (isValueChanged) {
+        this.dirty = true;
```

Comment:
**note:** The dirty state update is intentionally done before the validation call. In the following PR, `if(document.hasFocus())` will be replaced with `if(this.dirty)`.